core/src/tasks/recordProperty/importRecordProperties.ts

```diff
@@ -110,12 +110,46 @@ export class ImportRecordProperties extends RetryableTask {
 
     if (recordsToImport.length === 0) return;
 
-    let propertyValuesBatch: RecordPropertiesPluginMethodResponse = {};
     try {
-      propertyValuesBatch = await source.importRecordProperties(
+      const propertyValuesBatch = await source.importRecordProperties(
         recordsToImport,
         properties
       );
+
+      const orderedRecords: GrouparooRecord[] = [];
+      const orderedHashes: any[] = [];
+      for (const recordId in propertyValuesBatch) {
+        const record = recordsToImport.find((p) => p.id === recordId);
+        orderedRecords.push(record);
+        orderedHashes.push(propertyValuesBatch[recordId]);
+      }
+
+      if (orderedRecords.length > 0) {
+        await RecordOps.addOrUpdateProperties(
+          orderedRecords,
+          orderedHashes,
+          false // we are disabling the record lock here because the transaction will be saved out-of-band from the lock check
+        );
+      }
+
+      // update the properties that got no data back
+      await RecordProperty.update(
+        {
+          state: "ready",
+          rawValue: null,
+          stateChangedAt: new Date(),
+          confirmedAt: new Date(),
+        },
+        {
+          where: {
+            propertyId: { [Op.in]: properties.map((p) => p.id) },
+            recordId: {
+              [Op.in]: recordsToImport.map((p) => p.id),
+            },
+            state: "pending",
+          },
+        }
+      );
     } catch (error) {
```

Comment:
Prior to this PR only the import from the source (`source.importRecordProperties`) was in the try/catch.  Now we'll wrap the import and the saving of that data in the try/catch